node/core/approval-voting/src/lib.rs

```diff
@@ -2275,7 +2338,11 @@ async fn launch_approval<Context>(
 	validator_index: ValidatorIndex,
 	block_hash: Hash,
 	backing_group: GroupIndex,
+	span: &jaeger::Span,
 ) -> SubsystemResult<RemoteHandle<ApprovalState>> {
+	let mut launched_work_span = span
+		.child("launched-approval-work-future")
```

Comment:
The name might be missleading here. This only sets up the task and issues some requests to recover available data and validation code. These operations are usually really fast, but they can block if the `availability-recovery` subsystem queue is full.